core/src/services/icloud/core.rs

```diff
@@ -0,0 +1,648 @@
+                    Ok(())
+                }
+            }
+            _ => Err(Error::new(
+                ErrorKind::Unexpected,
+                "Apple iCloud AuthenticationFailed:Unauthorized:Invalid token",
+            )),
+        }
+    }
+}
+
+impl IcloudSigner {
+    pub async fn sign(
+        &mut self,
+        method: Method,
+        uri: String,
+        body: AsyncBody,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let mut request = Request::builder().method(method).uri(uri);
+
+        request = request.header(OAUTH_STATE_HEADER, self.data.oauth_state.clone());
+
+        if let Some(session_id) = &self.data.session_id {
+            request = request.header(SESSION_ID_HEADER, session_id);
+        }
+        if let Some(scnt) = &self.data.scnt {
+            request = request.header(SCNT_HEADER, scnt);
+        }
+
+        // China region
+        // ("Origin", "https://www.icloud.com.cn")
+        // ("Referer", "https://www.icloud.com.cn/")
+        if let Some(region) = &self.region {
+            request = request.header("Origin", region);
+            request = request.header("Referer", (region.to_string() + "/").as_str());
+        }
+
+        if !self.data.cookies.is_empty() {
+            let cookies: Vec<String> = self
+                .data
+                .cookies
+                .iter()
+                .map(|(k, v)| format!("{}={}", k, v))
+                .collect();
+            request = request.header(header::COOKIE, cookies.as_slice().join("; "));
+        }
+
+        if let Some(headers) = request.headers_mut() {
+            headers.insert("Content-Type", "application/json".parse().unwrap());
+            headers.insert("Accept", "*/*".parse().unwrap());
+            for (key, value) in AUTH_HEADERS {
+                headers.insert(key, value.parse().unwrap());
+            }
+        }
+
+        match self.client.send(request.body(body).unwrap()).await {
+            Ok(response) => {
+                if let Some(account_country) = response.headers().get(ACCOUNT_COUNTRY_HEADER) {
+                    self.data.account_country =
+                        Some(String::from(account_country.to_str().unwrap()));
+                }
+                if let Some(session_id) = response.headers().get(SESSION_ID_HEADER) {
+                    self.data.session_id = Some(String::from(session_id.to_str().unwrap()));
+                }
+                if let Some(session_token) = response.headers().get(SESSION_TOKEN_HEADER) {
+                    self.data.session_token = Some(String::from(session_token.to_str().unwrap()));
+                }
+
+                if let Some(scnt) = response.headers().get(SCNT_HEADER) {
+                    self.data.scnt = Some(String::from(scnt.to_str().unwrap()));
+                }
+
+                for (key, value) in response.headers() {
+                    if key == header::SET_COOKIE {
+                        #[warn(clippy::single_char_pattern)]
```

Comment:
Please remove this warning and chaning `";"` into `';'` instead.

Reply:
doit